src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -270,6 +270,27 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             // Note: persist "m_currentThread" to be compatible with old versions
             info.AddValue(ThreadIDField, 0);
         }
+        
+        [System.Diagnostics.Conditional("DEBUG")]
+        private void DebugDump()
+        {
+            if (Length == 0)
+            {
+                return;
+            }
+            StringBuilder chunk = this;
+            // showing at most the last 10 chunks in this StringBuilder
+            const int maxChunksToPrint = 10;
+            var debugText = new StringBuilder();
+            for (int i = 0; chunk != null && i < maxChunksToPrint; i++)
+            {
+                debugText.Insert(0, "|" + new string(chunk.m_ChunkChars));
```

Comment:
Ironically by doing Insert you are forcing StringBuilder to behave suboptimally...

You should be able to just Append repeatedly to add each char in the chunk.

Or if you want to get fancy (doesn't matter given it's debug code, but as a learning exercise) you can avoid creating a StringBuilder entirely and write directly into a string. See https://msdn.microsoft.com/en-us/magazine/mt814808.aspx  and search for text "Now, not only have you avoided the allocation, you’re writing directly into the string’s memory on the heap"